### PR TITLE
#35 Add blog posts #tags

### PR DESCRIPTION
**Pull request by ivorco**
This adds the tags to the posts page
However, I think this can be done better but not sure how
I need to also add the tags to the post [id].tsx page

### CODE REVIEW

**Comment by ivorco** [comment]
> Can you please add in `READMEE.md` the set up needed in contentful, i.e adding a tag content type, and linking posts to tags.

I've done better
Look at commit 17dab64c00ab5cf1f54aa315f0e085ecc053a19c

File: nextjs-website/README.md [hunk 1]
```diff
@@ -39,6 +39,10 @@ npm install
     - Find your `accessToken` in _Settings_ => _API keys_. You can use the example key generated by Conetntful or generate your own. Note that you are copying the _Content Delivery API - access token_.
     - Find your `environmentId` under _Settings_ => _Environments_.
 9. Remove the `example` suffix from the config file name. _The file should be ignored by `.gitignore`_.
+10. Import the contetful content model:
```
Comment (OPerel) [inline nextjs-website/README.md:42]: I guess this should happen after installation and login

File: nextjs-website/README.md [hunk 2]
```diff
@@ -69,3 +73,18 @@ To run the loading time tracing:
 npm run test:perf:trace
 ```
 
+## Export content model
+In order to export the content model you have got to install the contentful cli:
```
Comment (OPerel) [inline nextjs-website/README.md:77]: Move the setup instruction to the contentful set up section

File: nextjs-website/components/List.tsx [hunk 3]
```diff
@@ -18,12 +19,15 @@ const List: React.FunctionComponent<Props> = ({ posts }) => {
...
       {posts.filter(post => {
-        return  filterString(post.fields.title, filter) ||
-                filterString(post.fields.postText, filter) ||
-                filterString(post.fields.excerpt, filter);
+        return filterString(post.fields.title, filter) ||
+          filterString(post.fields.postText, filter) ||
+          filterString(post.fields.excerpt, filter);
       }).map(post => (
-        <ListItem key={post.sys.id} data={post} />
+        <ListItem key={post.sys.id} data={post} tags={post.fields.tags ? post.fields.tags.map((tag: any) => tags.find(tagType => tagType.sys.id == tag.sys.id).fields.tagName) : []} />
```
Comment (OPerel) [inline nextjs-website/components/List.tsx:30]: Please refactor the tag filtering to a function

File: nextjs-website/components/ListItem/ListItem.tsx [hunk 4]
```diff
@@ -4,16 +4,18 @@ import Link from 'next/link';
...
 
-const ListItem: React.FunctionComponent<Props> = ({ data }) => (
+const ListItem: React.FunctionComponent<Props> = ({ data, tags }) => (
   <div className={styles.item}>
     <Link href="/posts/[id]" as={`/posts/${data.sys.id}`}>
       <a><h3>{data.fields.title}</h3></a>
     </Link>
     <span>{new Date(data.sys.createdAt).toLocaleDateString()}</span>
     <p>{(data.fields.excerpt && data.fields.excerpt.trim()) ? data.fields.excerpt : data.fields.postText.split('\n')[0]}</p>
+    <p>{tags && tags.length > 0 ? "Tags: " + tags.map(tag => tag + " ") : ""}</p>
```
Comment (OPerel) [inline nextjs-website/components/ListItem/ListItem.tsx:18]: Why both `tags &&` and `tags.length`? isn't `tags.length > 0` enough?

File: nextjs-website/pages/posts/index.tsx [hunk 5]
```diff
@@ -40,12 +41,13 @@ export const getStaticProps: GetStaticProps = async () => {
...
   const accessToken = process.env.accessToken;
   const environmentId = process.env.environmentId;
-  const baseUrl = process.env.contentfulBaseUrl; 
+  const baseUrl = process.env.contentfulBaseUrl;
 
   try {
-    const allPosts = await fetch(`${baseUrl}/spaces/${spaceId}/environments/${environmentId}/entries?access_token=${accessToken}`);
-    const posts = await allPosts.json();
-    return { props: { posts: posts.items } }
+    const allPosts = await fetch(`${baseUrl}/spaces/${spaceId}/environments/${environmentId}/entries?access_token=${accessToken}&include=10`);
```
Comment (OPerel) [inline nextjs-website/pages/posts/index.tsx:47]: Here you can add another fetch call. One just for posts and one just for tags.

Comment (ivorco) [reply nextjs-website/pages/posts/index.tsx:47]: Why would I if I can get both in the same call?

Comment (OPerel) [reply nextjs-website/pages/posts/index.tsx:47]: Do whatever you think is more efficient. I was just referring to you saying that contentful gives you all the entries ([here](https://github.com/wonderfloyd/Hackathon-Corona/issues/35#issuecomment-616226386)), and wanted to say you can divide the calls to specific content types or entries if you need to.

Comment (ivorco) [reply nextjs-website/pages/posts/index.tsx:47]: Oh okay
I think I would push it as it is and then continue to making the [id] page
There I'll need a way to get the tags for a specific post so I'll build a code that can be used in both scenarios

**Pull request merged**